src/components/Project/Secrets/paginator.js

```diff
@@ -448,7 +461,29 @@ export default class SecretsPaginator extends React.Component {
           </Manager>
         </div>
 
-        <Drawer
+        {this.state.drawerOpen &&
+        <Query query={GET_SECRET} variables={{id: this.state.selectedSecretID}}>
+          {({ loading, error, data }) => {
+              if (loading) return <div></div>
+              if (error) return `Error! ${error.message}`
+              let secret = data.secret
+              if(secret.value !== "" && !self.form.isDirty && !this.state.saving){
```

Comment:
I'm not sure why you're using the `isDirty` flag here?

Reply:
As discussed - we do not want to re-initialize the form if the form values have been changed.